### PR TITLE
Add file type: SPM

### LINKED ISSUE #16: Migrate .spm support from TopoStats

### PR DESCRIPTION
**Pull request by SylviaWhittle**
Closes #16

This PR adds SPM support.

Just basic loading of `.spm` files, akin to how I did `.asd`.

### CODE REVIEW

File: topofileformats/spm.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    px_to_real = channel_data.pxs()
+    # Has potential for non-square pixels but not yet implimented
+    pixel_to_nm_scaling = (
+        px_to_real[0][0] * unit_dict[px_to_real[0][1]],
+        px_to_real[1][0] * unit_dict[px_to_real[1][1]],
+    )[0]
+    if px_to_real[0][0] == 0 and px_to_real[1][0] == 0:
+        pixel_to_nm_scaling = 1
+        print(f"[{filename}] : Pixel size not found in metadata, defaulting to 1nm")
```
Comment (ns-rse) [inline topofileformats/spm.py:35]: Perhaps...

```suggestion
        LOGGER.warning(f"[{filename}] : Pixel size not found in metadata, defaulting to 1nm")
```

File: topofileformats/spm.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    print(f"[{filename}] : Pixel to nm scaling : {pixel_to_nm_scaling}")
```
Comment (ns-rse) [inline topofileformats/spm.py:36]:
```suggestion
    LOGGER.info(f"[{filename}] : Pixel to nm scaling : {pixel_to_nm_scaling}")
```

File: topofileformats/spm.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+def load_spm(file_path: Path | str, channel: str) -> tuple:
+    """Extract image and pixel to nm scaling from the Bruker .spm file.
+
+    Returns
+    -------
+    tuple(np.ndarray, float)
+        A tuple containing the image and its pixel to nanometre scaling value.
+    """
+    print(f"Loading image from : {file_path}")
```
Comment (ns-rse) [inline topofileformats/spm.py:48]: This and others should probably be `LOGGER.[info|error|warning]()`

File: topofileformats/spm.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    except FileNotFoundError:
+        print(f"[{filename}] File not found : {file_path}")
+        raise
+    except Exception as e:
+        if "Channel" in str(e) and "not found" in str(e):
+            # trying to return the error with options of possible channel values
+            labels = []
+            for channel_option in [layer[b"@2:Image Data"][0] for layer in scan.layers]:
+                channel_name = channel_option.decode("latin1").split(" ")[1][1:-1]
+                # channel_description = channel.decode('latin1').split('"')[1] # incase the blank field raises quesions?
```
Comment (ns-rse) [inline topofileformats/spm.py:66]: Looks like this is left over, is the blank field likely to be an issue?

**Review by ns-rse — CHANGES_REQUESTED** [review]
Thanks @SylviaWhittle as with other PRs out some comments in-line.

Not sure what your plan is for integrating with TopoStats but we could likely use the "factory method" design pattern to select the correct loader based on file extensions.

File: pyproject.toml [hunk 5]
```diff
@@ -21,14 +21,16 @@ classifiers = [
     "Programming Language :: Python :: 3.8",
```
[the remark this answers is not in the document]
Comment (SylviaWhittle) [reply pyproject.toml:21]: Agree!

File: topofileformats/spm.py (hunk 4 above)
Comment (SylviaWhittle) [reply topofileformats/spm.py:70]: I can't remember why I wrote this. I'll revert it back to how it is in topostats.

**Review by ns-rse — APPROVED** [review]
:+1: Good to go.

**Pull request merged**